apps/sms/test/marionette/lib/messages.js

```diff
@@ -256,6 +256,34 @@
           client.helper.waitForElement(SELECTORS.main);
         },
 
+        launchFromNotification: function(notification) {
+          var chromeClient = client.scope({ context: 'chrome' });
+          chromeClient.executeScript(function(origin, notification) {
+            /* global Components, Services */
+            var managerClass = Components.classes[
+              '@mozilla.org/system-message-internal;1'
+            ];
+
+            var pageURI = Services.io.newURI(
+              origin + '/index.html#notification', null, null
+            );
+
+            var manifestURI = Services.io.newURI(
+              origin + '/manifest.webapp', null, null
+            );
+
+            var systemMessageManager = managerClass.getService(
+              Components.interfaces.nsISystemMessagesInternal
+            );
+
+            systemMessageManager.sendMessage(
+              'notification', notification, pageURI, manifestURI
+            );
+          }, [ORIGIN_URL, notification]);
+
+          this.switchTo();
+        },
+
```

Comment:
Well let me dump discussion I had with myself before :)

* I don't really think we should involve notification bar - it can change and it definitely will change and our app (and test) should not depend on it. So for me notification bar is basically like dialer or contacts - _external_ app/widget/component. Indeed e2e test you're describing is useful, but feels more like gaia-integration test (integration of apps residing somewhere in shared) rather than sms integration (integration of app views, no matter who called/run it). So I think about this test like "In manifest I see entry point for my app, I need to open it via this entry point and see what happens"

* "2 scenarios: one with killing the application - one with keeping it" - good tests, but in this PR I just want to test what I actually changed (going to conversation when thread list is not loaded), later we can add "one with keeping it" test as well.

Please tell me what you think :)